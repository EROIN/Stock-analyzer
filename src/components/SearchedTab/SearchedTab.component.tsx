import {useEffect, useState} from 'react';

import './SearchedTab.scss';

import {SearchedTabProps} from './SearchedTab.types';
import {TimeChart} from '../../components/TimeChart/TimeChart.component';
import {getTimeSeriesForSymbol} from '../../middleware/timeSeries.middleware';
import {GetTimeSeriesAPIResponse} from '../../types/timeSeries/timeSeries.types';
// import {TIME_SERIES_DATA} from '../../__mocks__/timeSeries.mocks';

export const SearchedTab = (props: SearchedTabProps) => {
  const {data, activeKey} = props;

  const [chartData, setChartData] = useState<GetTimeSeriesAPIResponse>();

  useEffect(() => {
    if (!chartData && activeKey && activeKey !== 'nodata')
      getTimeSeriesForSymbol(activeKey).then(setChartData).catch(console.log);
  }, [activeKey, chartData]);

  if (Object.keys(data).length == 0)
    return (
      <div className="search-tab-stock-name">Stock details not found.</div>
    );

  return (
    <div className="search-tab-container">
      <div className="search-tab-details-wrapper">
        <div className="search-tab-stock-name">
          {data.Name}
          &nbsp;<span>({data.Symbol})</span>
        </div>
        <div className="search-tab-stock-detail-row">
          Current Price: <span>{data['50DayMovingAverage']}</span>
          {/* (50day avg being used since API doesnt have price data) */}
        </div>
        <div className="search-tab-stock-detail-row">
          Exchange: <span>{data.Exchange}</span>
        </div>
        <div className="search-tab-stock-detail-row">
          Industry: <span>{data.Industry}</span>
        </div>
        <div className="search-tab-stock-detail-row">
          P/E: <span>{data.PERatio}</span>
        </div>
        <div className="search-tab-stock-detail-row">
          Market Cap:{' '}
          <span>
            {data.MarketCapitalization}
            &nbsp;{data.Currency}
          </span>
        </div>
        <div className="search-tab-mobile-chart">
          <TimeChart chartData={chartData} />
        </div>
        <div className="search-tab-stock-detail-row">
          <p>{data.Description}</p>
        </div>
      </div>
      <div className="search-tab-desktop-chart">
        <TimeChart chartData={chartData} />
      </div>
    </div>
  );
};

SearchedTab.defaultProps = {
  activeKey: '',
  data: {}, // added so that if someone doesnt pass by mistake, it doesnt break the page
};
