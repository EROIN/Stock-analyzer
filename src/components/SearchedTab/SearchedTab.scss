.search-tab-stock-name {
  font-size: 2em;
  font-weight: 700;

  span {
    font-size: 0.5em;
  }
}

.search-tab-stock-detail-row {
  font-weight: 600;

  span {
    font-weight: normal;
  }

  p {
    margin-top: 1em;
    font-family: Roboto;
    font-weight: 300;
  }
}

.search-tab-container {
  display: flex;
}

.search-tab-details-wrapper {
  flex: 1;
}

.search-tab-chart-container {
  width: 400px;
  height: 200px;
  align-self: center;
}

@media only screen and (max-width: 768px) {
  .search-tab-stock-name {
    font-size: 1.5em;
    margin-bottom: 0.5em;
  }

  .search-tab-container {
    margin-left: 0.5em;
  }
}
