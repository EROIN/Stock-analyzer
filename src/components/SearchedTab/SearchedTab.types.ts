import {StockDetailAPIResponse} from '../../types/search/symbolSearch.types';

export interface SearchedTabProps {
  data: StockDetailAPIResponse;
}
