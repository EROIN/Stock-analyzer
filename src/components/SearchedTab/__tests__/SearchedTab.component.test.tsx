import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import {SearchedTab} from '../SearchedTab.component';
// eslint-disable-next-line jest/no-mocks-import
import {SEARCH_RESULTS} from '../../../__mocks__/searchResult.mocks';

describe('<SearchedTab/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer
      .create(<SearchedTab data={SEARCH_RESULTS.results.IBM} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with empty data', () => {
    const tree = renderer
      //@ts-ignore
      .create(<SearchedTab data={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
