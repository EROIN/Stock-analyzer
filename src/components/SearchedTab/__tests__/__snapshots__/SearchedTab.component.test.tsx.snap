// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchedTab/> renders correctly with default props 1`] = `
<div
  className="search-tab-container"
>
  <div
    className="search-tab-details-wrapper"
  >
    <div
      aria-label="stock name is International Business Machines Corporation"
      className="search-tab-stock-name"
    >
      International Business Machines Corporation
       
      <span
        aria-label="stock symbol is IBM"
      >
        (
        IBM
        )
      </span>
    </div>
    <div
      className="search-tab-stock-detail-row"
    >
      Current Price: 
      <span>
        122.96
      </span>
    </div>
    <div
      className="search-tab-stock-detail-row"
    >
      Exchange: 
      <span>
        NYSE
      </span>
    </div>
    <div
      className="search-tab-stock-detail-row"
    >
      Industry: 
      <span>
        COMPUTER & OFFICE EQUIPMENT
      </span>
    </div>
    <div
      className="search-tab-stock-detail-row"
    >
      P/E: 
      <span>
        26.16
      </span>
    </div>
    <div
      className="search-tab-stock-detail-row"
    >
      Market Cap:
       
      <span>
        123889353000
         
        USD
      </span>
    </div>
    <div
      className="search-tab-mobile-chart"
    />
    <div
      className="search-tab-stock-detail-row"
    >
      <p>
        International Business Machines Corporation (IBM) is an American multinational technology company headquartered in Armonk, New York, with operations in over 170 countries. The company began in 1911, founded in Endicott, New York, as the Computing-Tabulating-Recording Company (CTR) and was renamed International Business Machines in 1924. IBM is incorporated in New York. IBM produces and sells computer hardware, middleware and software, and provides hosting and consulting services in areas ranging from mainframe computers to nanotechnology. IBM is also a major research organization, holding the record for most annual U.S. patents generated by a business (as of 2020) for 28 consecutive years. Inventions by IBM include the automated teller machine (ATM), the floppy disk, the hard disk drive, the magnetic stripe card, the relational database, the SQL programming language, the UPC barcode, and dynamic random-access memory (DRAM). The IBM mainframe, exemplified by the System/360, was the dominant computing platform during the 1960s and 1970s.
      </p>
    </div>
  </div>
  <div
    className="search-tab-desktop-chart"
  />
</div>
`;

exports[`<SearchedTab/> renders correctly with empty data 1`] = `
<div
  className="search-tab-stock-name"
>
  Stock details not found.
</div>
`;
